### ISSUE TITLE
Joining Deathmatch makes you DNR

### ISSUE THREAD

**Issue by OuroborusReporter** [tgstation/tgstation#88187]
### Client Version:

515.1637

### Issue Summary:

I didn't DNR, joined a deathmatch, and my brain went DNR
![Screenshot 2024-11-24 195141](https://github.com/user-attachments/assets/65fd651c-69e9-4986-8cfa-7062053521e5)

![Screenshot 2024-11-24 195127](https://github.com/user-attachments/assets/e768937b-baf9-4cb4-813c-1a0133bf5783)

### Round ID:

243377

### Test Merge Information:

- [Adds meowing and purring back for felinids](https://www.github.com/tgstation/tgstation/pull/87156)

### Reproduction Steps:

1. Die
1a. Possibly specifically have your brain removed from your body?
2. Join a deathmatch
2a. Possibly specifically CREATE a deathmatch lobby?
3. Become unrevivable

**Comment by mc-oofert** [comment]
did you happen to play with magic spells in that lobby

**Comment by OuroborusReporter** [comment]
> did you happen to play with magic spells in that lobby

Nope, it was maint brawl with all assistants. The only change was the modifier that supposedly makes you not fall into crit (?), which either didn't work or applied to a small enough health range that I didn't notice. No polymorphing of any kind

**Comment by mc-oofert** [comment]
I think the temporary body component just really hates bodyswaps and being a brain in particular

did you happen to be a brain before playing

**Comment by OuroborusReporter** [comment]
Yes, I was debrained (body gibbed, organs strewn across the cargo shuttle) before I went deathmatch

**Comment by mc-oofert** [comment]
> All brain transplants were broken. Should be fixed in #88198

irrelevant issue

**Comment by SmArtKar** [comment]
@Time-Green does not fix - this happens due to the component being unable to find a ghost when the body is qdeleted via chasms due to the component being unable to find a valid ghost

**Comment by Time-Green** [comment]
> @Time-Green your PR does not fix the issue - this happens due to the component being unable to find a ghost when the body is qdeleted via chasms due to the component being unable to find a valid ghost

Yeah I know I already removed it. I just didn’t bother to respond since oofert was mean

**Comment by mc-oofert** [comment]
> Time-Green your PR does not fix the issue - this happens due to the component being unable to find a ghost when the body is qdeleted via chasms due to the component being unable to find a valid ghost

cant get this to happen

**Comment by OuroborusReporter** [comment]
This got fixed at some point
